pkg/tls/tls.go

```diff
@@ -9,64 +9,100 @@ import (
 	"errors"
 	"net"
 	"os"
+
+	"github.com/pion/dtls/v2"
 )
 
 var (
-	errTLSdetails   = errors.New("failed to get TLS details of connection")
-	errLoadCerts    = errors.New("failed to load certificates")
-	errLoadServerCA = errors.New("failed to load Server CA")
-	errLoadClientCA = errors.New("failed to load Client CA")
-	errAppendCA     = errors.New("failed to append root ca tls.Config")
+	errTLSdetails     = errors.New("failed to get TLS details of connection")
+	errLoadCerts      = errors.New("failed to load certificates")
+	errLoadServerCA   = errors.New("failed to load Server CA")
+	errLoadClientCA   = errors.New("failed to load Client CA")
+	errAppendCA       = errors.New("failed to append root ca tls.Config")
+	errUnsupportedSec = errors.New("unsupported security configuration")
 )
 
-// Load return a TLS configuration that can be used in TLS servers.
-func Load(c *Config) (*tls.Config, error) {
+type SecConfig interface {
```

Comment:
Let's Rename this to something like

```suggestion
type TLSConfig interface {
```